pybamm/expression_tree/binary_operators.py

```diff
@@ -10,8 +11,15 @@
 import pybamm
 from pybamm.util import have_optional_dependency
 
+from typing import Union, Callable
 
-def _preprocess_binary(left, right):
+# create type alias(s)
+ChildValue = Union[float, np.ndarray, pybamm.Symbol]
```

Comment:
This is a bit pedantic, but I would probably call this `ChildSymbol` rather than a value. It only becomes a value when the tree is evaluated, and so you could have both, ie.:

```python
ChildSymbol = Union[float, np.ndarray, pybamm.Symbol]
ChildValue = Union[float, np.ndarray]
```